lib/nexpose/site.rb

```diff
@@ -430,23 +430,32 @@ def remove_user(user_id)
     end
 
     def self.from_hash(hash)
-      site = new(hash[:name], hash[:scan_template_id])
-      hash.each do |k, v|
-        site.instance_variable_set("@#{k}", v)
-      end
+      hash = Nexpose::SymbolizeHashKeys.symbolize(hash)
+      site = self.json_initializer(hash).deserialize(hash)
 
       # Convert each string address to either a HostName or IPRange object
-      included_scan_targets = { addresses: [], asset_groups: [] }
-      site.included_scan_targets[:addresses].each { |asset| included_scan_targets[:addresses] << HostOrIP.convert(asset) }
-      included_scan_targets[:asset_groups] = site.included_scan_targets[:asset_groups]
-      site.included_scan_targets = included_scan_targets
+      included_addresses = hash[:included_scan_targets][:addresses]
```

Comment:
This statement will throw an error if :included_scan_targets is not defined in the hash. You may wan to make this more resilient so that it can handle a hash where the caller doesn't specify a scan target, but you still initialize the hash yourself.